salt/roster/terraform.py

```diff
@@ -0,0 +1,237 @@
+    '''
+    primary = resource['primary']
+    attrs = primary['attributes']
+    name = attrs['name']
+    num_ifaces = int(attrs['network_interface.#'])
+    for if_k in range(num_ifaces):
+        num_addrs = int(attrs['network_interface.{}.addresses.#'.format(if_k)])
+        for addr_k in range(num_addrs):
+            addr = attrs['network_interface.{}.addresses.{}'.format(if_k, addr_k)]
+            if not addr:
+                continue
+            ret[name] = {'host': addr}
+            _add_ssh_key(ret[name])
+            return
+
+
+def _handle_aws_resource(resource, ret):
+    '''
+    Reads public_dns or public_ip attributes from AWS resources.
+    '''
+    primary = resource['primary']
+    attrs = primary['attributes']
+
+    addr = attrs['public_dns']
+    if not addr:
+        addr = attrs['public_ip']
+    if not addr:
+        return
+
+    name = attrs['tags.Name']
+    if not name:
+        name = attrs['id']
+    if not name:
+        return
+
+    ret[name] = {'host': addr}
+    _add_ssh_key(ret[name])
+
+
+def _add_ssh_key(ret):
+    '''
+    Setups the salt-ssh minion to be accessed with salt-ssh default key
+    '''
+    priv = None
+    if __opts__.get('ssh_use_home_key') and \
+       os.path.isfile(os.path.expanduser('~/.ssh/id_rsa')):
+        priv = os.path.expanduser('~/.ssh/id_rsa')
+    else:
+        priv = __opts__.get(
+            'ssh_priv',
+            os.path.abspath(os.path.join(
+                __opts__['pki_dir'],
+                'ssh',
+                'salt-ssh.rsa'
+            ))
+        )
+    if not priv:
+        return
+    if os.path.isfile(priv):
+        ret['priv'] = priv
+
+
+def _parse_state_file(state_file_path='terraform.tfstate'):
+    '''
+    Parses the terraform state file passing different resource types to the right handler
+    '''
+    ret = {}
+    with salt.utils.fopen(state_file_path, 'r') as fh_:
+        tfstate = json.load(fh_)
+
+    modules = tfstate.get('modules', None)
+
+    if not modules:
+        log.error('Malformed tfstate file. No modules found')
+        return ret
+
+    for module in modules:
+        resources = module['resources']
+        for _, resource in six.iteritems(resources):
+            if resource['type'] == 'libvirt_domain':
+                _handle_libvirt_resource(resource, ret)
+            if resource['type'] == 'aws_instance':
+                _handle_aws_resource(resource, ret)
+    return ret
+
+
+def targets(tgt, tgt_type='glob', **kwargs):  # pylint: disable=W0613
+    '''
+    Returns the roster from the terraform state file, checks opts for location, but defaults to terraform.tfstate
+    '''
+    roster_file = os.path.abspath('terraform.tfstate')
+    if __opts__.get('roster_file', None):
+        roster_file = os.path.abspath(__opts__['roster_file'])
+
+    if not os.path.isfile(roster_file):
+        log.error('Can\'t find terraform state file {}'.format(roster_file))
```

Comment:
Please use double-quotes and format with an index (no empty `{}`, since they won't work on Python 2.6).